packages/lodestar/src/network/network.ts

```diff
@@ -103,6 +105,52 @@ export class Libp2pNetwork extends (EventEmitter as { new(): NetworkEventEmitter
     await this.libp2p.hangUp(peerInfo);
   }
 
+  /**
+   * Connect to new peers given a subnet.
+   * @param subnet the subnet calculated from committee index
+   * @param inPeerIds peers already have this subnet
+   */
+  public async connectToNewPeersBySubnet(subnet: number, inPeerIds?: string[]): Promise<number> {
+    const peerIds = inPeerIds || [];
+    const discovery: Discv5Discovery = this.libp2p._discovery.get("discv5") as Discv5Discovery;
+    const discv5: Discv5 = discovery.discv5;
+    const peerInfosForSubnet: PeerInfo[] = await Promise.all(
+      discv5.kadValues()
+        .filter((enr: ENR) => enr.get("attnets"))
+        .filter((enr: ENR) => {
+          try {
+            return this.config.types.AttestationSubnets.deserialize(enr.get("attnets"))[subnet];
+          } catch (err) {
+            return false;
+          }
+        })
+        .map((enr: ENR) => enr.peerId().then((peerId) => {
+          const peerInfo = new PeerInfo(peerId);
+          peerInfo.multiaddrs.add(enr.multiaddrTCP);
+          return peerInfo;
+        })));
+    const peerInfos = peerInfosForSubnet.filter(peerInfo => !peerIds.includes(peerInfo.id.toB58String()));
```

Comment:
Might be good idea to break this function into smaller chucnk for readability.
Like you could extract this chunk to searchDiscv5Peers or something